src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RemoveEmptyFinalizersTests.cs

```diff
@@ -216,6 +218,86 @@ public class Class1
                     },
                 }
             }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestCode = @"
+Public Class Class1
+#If DEBUG Then
+	Protected Overrides Sub Finalize()
+		System.Diagnostics.Debug.Fail(""Finalizer called!"")
+    End Sub
+#End If
+End Class
+",
+                SolutionTransforms =
+                {
+                    (solution, projectId) =>
+                    {
+                        var project = solution.GetProject(projectId);
+                        var parseOptions = ((VisualBasicParseOptions)project.ParseOptions).WithPreprocessorSymbols(new KeyValuePair<string, object>("DEBUG", true));
+
+                        return project.WithParseOptions(parseOptions)
+                            .Solution;
+                    },
+                }
+            }.RunAsync();
+        }
+
+        [Fact, WorkItem(1241, "https://github.com/dotnet/roslyn-analyzers/issues/1241")]
+        public async Task CA1821_DebugFailAndReleaseDirective_NoDiagnostic_FalsePositive()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+public class Class1
+{
+#if RELEASE
+	// There should be a diagnostic because in release the finalizer will be empty.
+	~Class1()
+	{
+		System.Diagnostics.Debug.Fail(""Finalizer called!"");
+    }
+#endif
+}
+",
+                SolutionTransforms =
+                {
+                    (solution, projectId) =>
+                    {
+                        var project = solution.GetProject(projectId);
+                        var parseOptions = ((CSharpParseOptions)project.ParseOptions).WithPreprocessorSymbols("DEBUG");
```

Comment:
😕 Is this supposed to be `RELEASE`?

Reply:
Yes, this should be `RELEASE`. Maybe the test should be skipped and contain the expected diagnostic. The idea of the test was to show that we currently disregard the actual symbol matching between the `#if` directive and the `ConditionalAttribute`. Another example would be to have `#if NET_20` instead of `#if RELEASE` and the finalizer containing a call to a method marked with the conditional let's say `NETCORE_APP_3.1` there wouldn't be any diagnostic raised while it should.